src/Strategy/Haskell/Cabal.hs

```diff
@@ -109,47 +109,47 @@ cabalGenPlanCmd =
 cabalPlanFilePath :: Path Rel File
 cabalPlanFilePath = $(mkRelFile "dist-newstyle/cache/plan.json")
 
-isCabalFile :: Path Abs File -> Bool
-isCabalFile file = isDotCabal || isCabalDotProject
-  where
-    name = fileName file
-    isDotCabal = ".cabal" `isSuffixOf` name
-    isCabalDotProject = "cabal.project" == name
-
 discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Exec rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
 discover dir = context "Cabal" $ do
   projects <- context "Finding projects" $ findProjects dir
   pure (map mkProject projects)
 
+isDotCabal :: Path Abs File -> Bool
+isDotCabal name = ".cabal" `isSuffixOf` fileName name
+
+isCabalDotFile :: Path Abs File -> Bool
+isCabalDotFile name = "cabal.project" == fileName name
+
 findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [CabalProject]
 findProjects = walk' $ \dir _ files -> do
-  let project =
-        CabalProject
-          { cabalDir = dir
-          }
+  let dotCabalFiles = filter isDotCabal files
+  let cabalDotFiles = filter isCabalDotFile files
+  let allDotFiles = dotCabalFiles ++ cabalDotFiles
 
-  if any isCabalFile files
-    then pure ([project], WalkSkipAll)
+  -- TODO: @wes is one preferred over the other if they are *both* found?
```

Comment:
@scruffystuffs - question for you since it looks like you may have built this. If there is both a `*.cabal` file AND a `cabal.project` file, I *assume* we should just be setting the origin path here to the `*.cabal` file (since we specify our deps in the `spectrometer.cabal` file. Wanted to confirm this behavior with you.

For more context, this field will be used to signal to users *where* we found deps (i.e. what manifest file)